peer/peerstorage_test.go

```diff
@@ -0,0 +1,69 @@
+package peer
+
+import (
+	"os"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/stretchr/testify/require"
+)
+
+// TestKvdbPeerStorage tests the KvdbPeerStorage functionality.
+func TestKvdbPeerStorage(t *testing.T) {
+	rt := require.New(t)
+
+	file, err := os.CreateTemp("", "*.db")
+	rt.NoError(err, "unable to create temp file")
+	t.Cleanup(func() {
+		rt.NoError(file.Close())
+		rt.NoError(os.Remove(file.Name()))
+	})
+
+	dbPath := file.Name()
+	db, err := kvdb.Open(
+		kvdb.BoltBackendName, dbPath, true, kvdb.DefaultDBTimeout,
+	)
+	t.Cleanup(func() {
+		rt.NoError(db.Close())
+	})
+
+	rt.NoError(err, "unexpected err creating kvdb DB")
+
+	// Create a KvdbPeerStorage instance.
+	peerDataStore, err := NewKvdbPeerStorage(db)
+	rt.NotNil(peerDataStore)
+	rt.NoError(err, "unexpected error creating new "+
```

Comment:
i'd swap these since if peerDataStorage _is_ nil, it means that the `err` is not nil & knowing that will provide us with more info regarding what went wrong. 

In general - i dont think you need the `NotNil` check if you have already checked that there is no error.

Reply:
> In general - i dont think you need the NotNil check if you have already checked that there is no error.

While it is idiom for non-nil error to imply nil pointer, there is nothing redundant about these checks in the testing context. The property you are depending on to say it is redundant is one that needs to be ensured, and this check in the test does exactly that. IRL we don't need to check both, but in the test I support both assertions being here.